### PR TITLE
Connectivity Adapter: Generating certs with token 

### PR DESCRIPTION
**Pull request by akgalwas**
**Description**

Changes proposed in this pull request:

- Generate certificate with token (/v1/applications/signingRequests/info, /v1/applications/management/info, /v1/applications/certificates endpoints)

### CODE REVIEW

**Comment by akgalwas** [comment]
The following functionality will be implemented in the next step:
- Certificate renewal
- Certificate revocation

There are also some things not included in this PR:
- Setting tenant in ClientIdentity returned from /management/info
- Setting eventInfoURL in /management/info and signingRequests/info
- Getting events Base URL from Director

File: components/connectivity-adapter/internal/connector/api.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	CompassConnectorURL string `envconfig:"default=http://compass-connector.compass-system.svc.cluster.local:3000/graphql"`
+	AdapterBaseURL      string `envconfig:"default=https://adapter-gateway.kyma.local"`
+	AdapterMtlsBaseURL  string `envconfig:"default=https://adapter-gateway-mtls.kyma.local"`
+}
+
+const (
+	timeout = 30 * time.Second
+)
+
+func RegisterHandler(router *mux.Router, config Config) error {
```
Comment (pkosiec) [inline components/connectivity-adapter/internal/connector/api.go]: Consider structuring the code similarly to `appregistry.go` - I think all routes in single place improve readability of the code. But it's just a suggestion.

File: chart/compass/charts/connectivity-adapter/templates/deployment.yaml [hunk 2]
```diff
@@ -25,6 +25,14 @@ spec:
             - name: http
               containerPort: {{ .Values.global.connectivity_adapter.port }}
               protocol: TCP
+          env:
+            - name: APP_CONNECTOR_COMPASS_CONNECTOR_URL
+              value: http://compass-connector.compass-system.svc.cluster.local:3000/graphql
```
Comment (pkosiec) [inline chart/compass/charts/connectivity-adapter/templates/deployment.yaml]: You can use values from Chart - such as Connector origin:
```
"http://compass-connector.{{ .Release.Namespace }}.svc.cluster.local:{{ .Values.global.connector.graphql.external.port }}"
```

File: components/connectivity-adapter/internal/connector/graphql/client.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
+package graphql
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply components/connectivity-adapter/internal/connector/graphql/client.go]: As agreed I will leave it as is for now

**Comment by akgalwas** [comment]
> After some manual testing I found some differences in the Adapter in comparison to Application Connector:
> 
> * no `eventsInfoUrl`s content provided in `/v1/applications/signingRequests/info` and `/v1/applications/management/info`
> * no response body from `/v1/applications/signingRequests/info` if code is `403`
> * no response body from `/v1/applications/certificates` if code is `403`
> * different code and response body from `/v1/applications/certificates` if wrong CSR is provided (code `500` instead of `400` and code `1` in the response body instead of `400`)

I will address this issues in the next step.

**Pull request merged**